### ISSUE TITLE
About the Selection of Exemplar Images on Geode Dataset

### ISSUE THREAD

**Issue by lh218** [facebookresearch/Contextualized-Vendi-Score-Guidance#1]
Thank you very much for your excellent work! I am eager to reproduce the numerical experiments presented in your paper. However, I've encountered some confusion regarding the selection of exemplar images. Specifically, in the **"generate.py" file, line 61** reads:

**real_hold_not_in_eval_sub = pd.read_csv('real_hold_not_in_eval_sub.csv')**

Despite reviewing the comments for this line, I'm still uncertain about the structure of this CSV file. Would it be possible for you to provide an example CSV file that would allow me to reproduce the results from your paper? This would be immensely helpful in my efforts to replicate your experiments accurately.

**Comment by ReyhaneAskari** [comment]
yes, the csv file basically contains some random examples that are not part of the evaluation set. I think checking the DIG-In library should help: https://github.com/facebookresearch/DIG-In/tree/main

I also checked now to see what we have in the columns and this is the list of columns:
```
>>> real_hold_not_in_eval_sub.columns
Index(['file_path', 'object', 'region', 'ip_country', 'date', 'make', 'model',
       'gps_position', 'gps_altitude', 'resolution', 'licence_plate',
       'people_in_background', 'tree_tag', 'features', 'id', 'str_features',
       'r', '_merge'],
      dtype='object')
```

We do not use all the columns, mostly the `file_path`, `object` and `region`. Hope this helps!

**Comment by lh218** [comment]
Thank you for your reply! I have reviewed the DIG-In codebase at https://github.com/facebookresearch/DIG-In/tree/main and encountered an issue. The codebase does not provide instructions on how to obtain two crucial files required in **compute_metric.py**:

**geode_all_inceptionv3.pkl (referenced on line 92)**
**dollarstreet_all_inceptionv3.pkl (referenced on line 108)**

I noticed this issue has been previously reported but remains unresolved. Therefore, I ask for your help about the following two questions:

How can I obtain these .pkl files to run the evaluation code?
How can I generate the real_hold_not_in_eval_sub.csv file, which is also missing from the documentation?

Any guidance on these matters would be greatly appreciated.

**Comment by melissah14** [comment]
Hi @lh218, thanks for reaching out! I collaborated with @ReyhaneAskari on this C-VSG project and also worked on the DIG-In project so wanted to provide some additional context here. 

I've now updated the DIG-In [Issue](https://github.com/facebookresearch/DIG-In/issues/1) to address how the `.pkl` files are formed (GeoDE and DollarStreet follow the same process). Please let me know if there is still uncertainty remaining with this additional information. 

For additional context regarding the selection of exemplar images for C-VSG -- In the DIG-In paper and in this work, we take an additional step of sub-sampling the GeoDE dataset and generated images so that there is exactly the same number of images for each object-region combination. This is to ensure a "fair" comparison across objects and regions, since different quantities of images correspond to different manifold sizes and can impact measurements. In order to get this amount, we had to remove some objects that had too few samples and also remove extraneous images of objects that had many samples. Thus, we ended up only include 29160 (180 images * 6 regions * 27 objects)  images overall in our results. This step of filtering is done here: https://github.com/facebookresearch/DIG-In/blob/main/compute_metrics.py#L258.

We use the remaining images that were *not* included in the sub-sampling for evaluation as the exemplar images for the C-VSG experiments. We stratified by object (and region, for some experiments) that correspond to a given prompt and select from the remaining images randomly in order to select the exemplar images.
